boat_utilities/src/boat_utilities/angles.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+# NOTE: All operations in this module use degrees CCW from East, as in a unit circle
+# TODO: Switch to rads
+__UNITS = 360.0
+
+
+def cosd(angle):
+    """Calculate the cosine of the following angle, in degrees."""
+    from math import cos, radians
+    return cos(radians(angle))
+
+
+def sind(angle):
+    """Calculate the sine of the following angle, in degrees."""
+    from math import sin, radians
+    return sin(radians(angle))
+
+
+def tand(angle):
+    """Calculate the tangent of the following angle, in degrees."""
+    from math import tan, radians
+    return tan(radians(angle))
+
+
+def atan2d(y, x):
+    """Calculate the arctan of the following angle, in degrees."""
+    from math import atan2, degrees
+    return degrees(atan2(y, x))
+
+
+def normalize(angle):
+    """Normalize the angle to be 0 to 360."""
+    from math import fmod
+    return fmod(fmod(angle, __UNITS) + __UNITS, __UNITS)
+
+
+def normalize_signed(angle):
+    """Normalize the angle to be -180 to 180."""
+    angle = normalize(angle)
+    if angle > __UNITS/2:
+        angle -= __UNITS
+    return angle
+
+
+def is_within_bounds(val, lower, upper):
```

Comment:
Depending on the answer to the previous comment about the bidirectional `is_within_bounds` i think maybe it makes sense to make another one that is bidirectional, or replace this altogether.

Reply:
`is_within_bounds` used to simply check if a value was between two bounds. After looking at our code, we only ever used this when considering angles. Thus, I moved this func into the `angles` module, where we treat the inputs as angles. However, since we treat angles as continuous, we *must* specify a lower and upper bound. If they were interchangeable, then everything would always be within the bounds in one direction or the other, since we're on a continuous circle.

Reply:
This is true, although we could take the segment of the unit circle that is less than 180 as the bound.  I'm good to leave this function as is, as long as that one TODO you made is not actually an issue.  Maybe that one requires a special method though.